tensorrt_llm/_torch/models/modeling_speculative.py

```
@@ -1137,6 +1174,19 @@ def _build_fused_kv_buffers(self) -> None:
         kv_weights = [
             a.qkv_proj.weight[q_size:q_size + 2 * kv_size] for a in layers_attn
         ]
+        # Fold each drafter layer's input_layernorm weight into its KV projection
+        # so context K/V match the query path. vLLM laguna_dflash applies
+        # layer.input_layernorm to context states before KV; RMSNorm gives
+        # (x_hat * w) @ Wkv.T == x_hat @ (Wkv * w).T, and the shared 1/rms(x) is
+        # applied to projected_hidden in precompute_context_kv.
+        dlayers = self.model.layers
+        if all(hasattr(dl, 'input_layernorm') for dl in dlayers):
+            kv_weights = [w * dl.input_layernorm.weight.data[None, :].to(w.dtype)
+                          for w, dl in zip(kv_weights, dlayers)]
```

Ruff (0.15.18)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)